midealocal/devices/c3/__init__.py

```diff
@@ -290,11 +290,11 @@ def set_target_temperature(
         self,
         target_temperature: float,
         mode: int | None,
-        zone: int,
+        zone: int | None = None,
     ) -> None:
         """Midea C3 device set target temperature."""
         message = self.make_message_set()
-        if self._attributes[DeviceAttributes.zone_temp_type][zone]:
+        if zone and self._attributes[DeviceAttributes.zone_temp_type][zone]:
```

Comment:
What happens if zone == 0?

Reply:
is 0 a valid value ? I think zone is >=1 or None

Reply:
Obviously `zone` is 0 or 1

```python
            message.zone_target_temp[zone] = target_temperature
```

```python
            if zone == 0:
                message.zone1_power = True
            else:
                message.zone2_power = True
```

```python
        zone1_target_temp = int(self.zone_target_temp[0])
        zone2_target_temp = int(self.zone_target_temp[1])
```

Reply:
> Obviously `zone` is 0 or 1

or None ;-)

Reply:
Is enough to change the line to "if zone is not None..."

Reply:
#180

Reply:
`None` is not valid value, maybe should raise an error